lib/router/handler-info/unresolved-handler-info-by-object.js

```diff
@@ -25,7 +25,7 @@ var UnresolvedHandlerInfoByObject = subclass(HandlerInfo, {
   serialize: function(_model) {
     var model = _model || this.context,
         names = this.names,
-        serializer = this.serializer || (this.handler && this.handler.serialize);
+        serializer = this.serializer || (this.handler && this.handler.serialize && this.handler.serialize.bind(this.handler));
```

Comment:
That's the way JavaScript works. When you get a reference to a method, `this` in it is not bound. The same applies for `this.serializer`, but I couldn't figure out where you use/set that and that's why I left it unmodified.